### PR TITLE
Prevent non-UTF8 user_agents from being slog'ed.

### PR DESCRIPTION
**Pull request by ashercodeorg**
Attempts to fix [Honeybadger error](https://app.honeybadger.io/projects/3240/faults/30788473).

### CODE REVIEW

File: dashboard/app/controllers/api_controller.rb [hunk 1]
```diff
@@ -206,7 +206,7 @@ def user_progress_for_stage
         script_level_id: script_level.try(:id),
         level_id: level.id,
         user_agent: request.user_agent,
-        locale: locale
+        locale: locale.valid_encoding? ? locale : 'invalid_encoding'
```
Comment (ashercodeorg) [inline dashboard/app/controllers/api_controller.rb]: Is it worth trying to determine the proper encoding and forcing it to UTF-8?

File: dashboard/app/controllers/api_controller.rb [hunk 2]
```diff
@@ -206,7 +206,7 @@ def user_progress_for_stage
         script_level_id: script_level.try(:id),
         level_id: level.id,
         user_agent: request.user_agent,
```
Comment (joshlory) [inline dashboard/app/controllers/api_controller.rb]: Should we add the same defensive fix here, if it can contain invalid characters?

Comment (ashercodeorg) [reply dashboard/app/controllers/api_controller.rb]: My inclination is that the non-UTF-8 characters are in the locale, not the user_agent. Should this not resolve the issue, I'll try the fix on user_agent. However, I'd prefer to not sanitize stuff needlessly.

**Comment by ashercodeorg** [comment]
Good catch, Will, my intuition is clearly wrong. Fixing...

**Comment by ashercodeorg** [comment]
Good to merge now?

**Pull request merged**